google/cloud/forseti/scanner/scanners/iam_rules_scanner.py

```diff
@@ -32,6 +35,58 @@
 LOGGER = logger.get_logger(__name__)
 
 
+def _add_bucket_ancestor_bindings(policy_data):
+    """Add bucket relevant IAM policy bindings from ancestors.
+
+    Resources can inherit policy bindings from ancestors in the resource
+    manager tree. For example: a GCS bucket inherits a 'objectViewer' role
+    from a project or folder (up in the tree).
+
+    So far the IAM rules engine only checks the set of bindings directly
+    attached to a resource (direct bindings set (DBS)). We need to add
+    relevant bindings inherited from ancestors to DBS so that these are
+    also checked for violations.
+
+    If we find one more than one binding with the same role name, we need to
+    merge the members.
+
+    NOTA BENE: this function only handles buckets and bindings relevant to
+    these at present (but can and should be expanded to handle projects and
+    folders going forward).
+
+    Args:
+        policy_data (list): list of (parent resource, iam_policy resource,
+            policy bindings) tuples to find violations in.
+    """
+    storage_iam_roles = frozenset([
+        'roles/storage.admin',
```

Comment:
These values should not be hard-coded.  We should allow users to define the roles in the rules file.

Reply:
I don't understand.

Reply:
There could be other roles that can grant access to gcs buckets: dataflow, dataproc, iam, etc.  So, by putting this in the config file, at least the user can see what roles we are enforcing on, and they can tweak if necessary.

Or, by hard-coding here, are we okay to basically only support scanning on the storage role?  And how do we let users know?